Deprecated `/Modules/FluidProperties/*` syntax

joshuahansel opened the pull request:
Refs #22093

joshuahansel committed:
Deprecated Modules/FluidProperties/* syntax

Refs #22093

joshuahansel committed:
Updated fluid properties syntax in tests and docs

Refs #22093

joshuahansel commented:
Note to reviewers: The first commit has the good stuff, and the second commit is all of the input file fixes. For the second commit, I recommend just looking at the result of the deprecated test target to make sure none of them are fluid property syntax deprecated errors.

lindsayad commented on modules/fluid_properties/doc/content/source/actions/AddFluidPropertiesDeprecatedAction.md:
```diff
@@ -0,0 +1,5 @@
+# AddFluidPropertiesDeprecatedAction
```
Do we not have action deprecation macros like we do for moose objects?

joshuahansel replied:
I don't know - where might those be defined (or where are the moose object ones defined)?

GiudGiud replied:
in base/Registry.h

lindsayad replied:
sadly it looks like we don't currently

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1174841/) on a17a21b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22109/site)

This comment will be updated on new commits.

lindsayad commented:
Mention me when you want more review/tests are passing 😄

GiudGiud commented:
@lindsayad
the framework part is ready. 
Test failures are trivial here, just some moved files that need an updated path, they dont preclude reviewing in my opinion

GiudGiud commented:
oh my bad wrong PR to comment on

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1177006/) on 0921e85 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a9629f</th>
      <th colspan=4>#22109 0921e8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.20%</td>
      <td>84.20%</td>
      <td>-0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>79337</td>
      <td>79332</td>
      <td>-5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14885</td>
      <td>14890</td>
      <td>+5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22109/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22109/coverage/framework)


## Modules coverage

### Fluid properties

<table>
  <thead>
    <tr>
      <th></th>
      <th>a9629f</th>
      <th colspan=4>#22109 0921e8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.44%</td>
      <td>85.36%</td>
      <td>-0.08%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>5643</td>
      <td>5643</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>962</td>
      <td>968</td>
      <td>+6</td>
      <td>6</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22109/coverage_diff/fluid_properties)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22109/coverage/fluid_properties)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/richards)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22109/coverage/xfem)

</details>

## Warnings

- `fluid_properties` new line coverage rate 0.00% is less than the suggested 90.0%



This comment will be updated on new commits.

joshuahansel commented:
I don't know what's going on with this doc failure:
```
ERROR: Unable to locate local heading with URL 'AddFluidPropertiesAction.html'
/data/civet2/build/moose/modules/fluid_properties/doc/content/source/actions/AddFluidPropertiesDeprecatedAction.md:1
 ┌─────────────────────────────────────────────────────────────────────────────────────────────────┐
1│[AddFluidPropertiesAction.md]                                                                    
```
I build it locally and don't get this. I'll try invalidating 🤷‍♂️

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1176973/) on 0921e85 : invalidated by @joshuahansel

joshuahansel commented:
That didn't work. I'll just avoid that link on that page, since it's a deprecated page anyway...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1179128/) on 033340f wanted to post the following:

The following commit(s) contain fixups:

- 033340f63bc: fixup! Deprecated Modules/FluidProperties/* syntax

GiudGiud reviewed (changes requested):
700 clicks later..

joshuahansel commented on modules/fluid_properties/doc/content/syntax/FluidProperties/index.md:
```diff
@@ -0,0 +1,39 @@
+| `rho` | $\rho$ | Density |
+| `s` | $s$ | Specific entropy |
+| `T` | $T$ | Temperature |
+| `v` | $v$ | Specific volume |
```
I don't know what you mean - I copied this file from `syntax/Modules/FluidProperties/index.md`, since that file will be deleted in favor of this one.

joshuahansel replied:
I'll use `!include` in the deprecated `index.md`

joshuahansel replied:
The app doc failures correspond to apps that have already been replacing the syntax, and they get a recursive documentation include because they had their `syntax/FluidProperties/index.md` links to `syntax/Modules/FluidProperties/index.md`, which now links to `syntax/FluidProperties/index.md`...

Rather than break apps, I'm just going to duplicate the content of `syntax/FluidProperties/index.md` to `syntax/Modules/FluidProperties/index.md` to avoid that link.

joshuahansel commented on modules/fluid_properties/doc/content/syntax/FluidProperties/index.md:
```diff
@@ -0,0 +1,39 @@
+
+## Interfaces
+
+Because two independent, intensive thermodynamic properties define a
+thermodynamic state, most interfaces in these objects are of the form $f(a,b)$
+where $f$ is the desired thermodynamic property and $a$ and $b$ are independent,
```
Yes, I can improve this page while I'm here.

joshuahansel replied:
I started this improvement and am getting carried away with it. I'll do a separate pull request for those changes, since it's bound to hold things up.

joshuahansel commented on modules/fluid_properties/src/actions/AddFluidPropertiesDeprecatedAction.C:
```diff
@@ -0,0 +1,27 @@
+  : AddFluidPropertiesAction(params)
+{
+  mooseDeprecated(
+      "The syntax 'Modules/FluidProperties/*' is deprecated. Use 'FluidProperties/*' instead.");
```
You forgot to delete the first "syntax" so I won't commit that suggestion :P But I'll change the wording

joshuahansel committed:
Avoided recursive link in syntax/Modules/FluidProperties/index.md

For the syntax change Modules/FluidProperties/* to FluidProperties/*,
some apps are already replacing the syntax, and their documentation
for that syntax is referring to this now-deprecated syntax file,
which was referring back to the non-deprecated file.
The content was duplicated, but this file will be deleted in the
future.

Refs #22093

GiudGiud reviewed (approved):
good clean up!
I ll update VTB when it s through
